src/box/lua/config/utils/schema.lua

```diff
@@ -18,6 +18,65 @@ local schema_mt = {}
 
 local scalars = {}
 
+-- {{{ Array helpers
+
+-- Convert array to string
+local function arr2str(array)
+    if type(array) ~= "table" then return tostring(array) end
+    local result = "["
+    for i, v in ipairs(array) do
+        if type(v) == "table" then
+            result = result .. arr2str(v)
+        else
+            result = result .. tostring(v)
+        end
+        if i < #array then
+            result = result .. ", "
+        end
+    end
+    result = result .. "]"
+    return result
+end
```

Comment:
I suppose you should reuse the existing implementation of lua formatter. We have `internal.format_lua`, you can look into `src/box/lua/console.lua` for inspiration.

Reply:
I vote for using the JSON encoder for prettifying data in an error/log message. We already do it here and there.

Also, in my opinion this Lua formatter is barely ready for production use: it was written once and we never touch it again. There is a couple of [known problems](https://github.com/tarantool/tarantool/issues?q=is%3Aopen+is%3Aissue+label%3A%22lua+output%22). (To be precise, @mandesero looked over them and reported that some of them are not reproducing anymore. However, this is likely a side effect of some other fixes -- there were no direct effect toward a good Lua formatter.)

I agree in your general point: we shouldn't write a formatter in place for a complex arbitrary data structure. Let's use something already implemented and widely used.